### ISSUE TITLE
In-build dual voice for sapi 5 in NVDA 2019.3 and above

### ISSUE THREAD

**Issue by artimei** [nvaccess/nvda#10740]
Hello. I think that everyone knows about such a wonderful and unique addition as Dual voice for automatic switching of voices to sapi 5.
Unfortunately, the developer has long ceased support for the addon and does not get in touch. since 2019.3 requires Python 3, this addon does not work anymore with new versions of nvda.
why not make support for automatic sapi 5 voice switching inside nvda? especially since the source code of the addon is.
I am Russian and use the Ivona synthesizer: Tatyana for the Russian text and Sally for English. I tried all available synthesizers that exist and came to the conclusion that not one Russian voice of any synthesizer reads normally English text. they read, but it’s very difficult to understand what exactly. also Tatyana almost perfectly pronounces my native language, which other synthesizers cannot afford. therefore, the lack of this synthesizer in future versions of nvda is a problem for me. I think that not only for me.

**Comment by josephsl** [comment]
Hi,

I think we are talking about two different things here:

1. Dual Voice add-on: at this time, this add-on is listed as incompatible. The best option now is contacting the author one more time.
2. Ivona voices: I advise contacting the synthesizer developer about this one.

I'm in favor of closing this issue as "wontfix" unless other people say it should be investigated.

Thanks.

**Comment by Adriani90** [comment]
However, if this is especially an issue with Sapi 5 voices and auto switching language, at least NVDA should take care of the language switching feature at driver level and not at synthesizer level. However, I do not have experience that much with SAPI 5 voices and cannot really assess if there are related bugs at driver level.
cc: @michaelDCurran

**Comment by artimei** [comment]
The Ivona synthesizer works like sapi 5 in nvda, like any other synthesizer, but only as one voice. without the ability to automatically switch voices. This function is performed by a two-voice addon.
The addon developer does not contact.
if you connect a synthesizer in the form of an add-on, such as a vocalizer, then automatic switching will work. but the Ivona synthesizer does not have an addon.

**Comment by josephsl** [comment]
Hi, in this case, the best option going forward would be letting the NVDA community take care of Dual Voice add-on. I’ll go ahead and ask on the appropriate mailing list. Thanks.

**Comment by Adriani90** [comment]
Ok then it seems the problem is not at driver level but on synth level.
Note that there are more than one Ivona voice vendors on the market. Did you try to contact all of them?
http://www.ttsforaccessibility.com/contact_tts/
https://www.readspeaker.com/support/
https://harposoftware.com/en/contact

And by the way, the company belongs to Amazon since 2014. So you might contact the amazon development center to address your issues with the voices.

Since Ivona is a paid license product, it will not be implemented in NVDA.

Closing as invalid.

**Comment by Mahmood-Taghavi** [comment]
Hi everyone,
I am the developer of dual voice add-on.
The problem is not upgrading to Python 3. It seems the NVDA synthesizer code is re-designed. Hence, I could not add sliders for control speed and other parameters of the second voice in the NVDA voice setting dialog. I tried to design an attachable dialog to NVDA using WX which be similar to the vocalizer synthesizer setting dialog to support the setting of second voice such as voice name, speed, pitch, and volume. Unfortunately, another problem seems to exist in the NVDA re-designed Sapi 5 code (or other dependent codes) which not let using SAPI 5 XML to change voice, pitch, volume, and rate (actually it lead to errors). Hence, I could not use Sapi 5 XML and cannot make dual voice add-on compatible with the upcoming NVDA until the NVDA fix that recent serious problem with SAPI 5 XML.
Moreover, I prefer the voice changing support internally be implemented in NVDA (if possible) because the dual voice could switch only between SAPI 5 voice while some users wanted to use for example two voices from different bases such as Microsoft core and SAPI5 (or eSpeak) which is not easy to implement in an addon. Furthermore, I wish NVDA provides some statistics about statistics of add-on active users for developers and perhaps with complementary information such as their setting for NVDA, also if users could request features and vote for it, it would show the quantity of the importance of each request features to the NVDA community. 
Best
Mahmood Taghavi
https://github.com/Mahmood-Taghavi/dual_voice

**Comment by Adriani90** [comment]
Thanks very much @Mahmood-Taghavi for your contribution. I am reopening this discussion.
cc: @michaelDCurran, @codeofdusk

**Comment by codeofdusk** [comment]
Looks like a duplicate of #279 to me. Should be much easier to implement now thanks to speech refactor.

**Comment by Adriani90** [comment]
Thanks @codeofdusk, good pick. I also think this is a duplicate. @Mahmood-Taghavi please follow the discussion in #279.

**Comment by Mahmood-Taghavi** [comment]
Hi dear users and developers.
I am pleased to announce a new release of the Dual Voice addon for NVDA which is now compatible with the NVDA 2019.3.1.
Dual voice is an addon that made possible use of two different voices (currently both voices must support SAPI5 standard) for reading two languages one with Latin script and another with non-Latin writing script. For example, a user can select a voice for reading English as a language with the Latin writing script and select a voice for reading Persian (my language) as a language with the non-Latin writing script. 
Languages with the Latin writing script are English, Czech, Croatian, Finnish, French, German, Italian, Polish, Portuguese, Slovenian, Spanish, and Turkish.
Languages with the non-Latin script are Persian, Arabic, Belarusian, Bulgarian, Chinese, Greek, Hebrew, Japanese, Korean, Russian, and Ukrainian.
You can find more information on the [homepage of the Dual Voice for NVDA](https://mahmood-taghavi.github.io/dual_voice/).
You can download the [latest version of the Dual Voice for NVDA](https://github.com/Mahmood-Taghavi/dual_voice/releases/download/v4.1/dual_voice-4.1.nvda-addon)
Note: You can now use a custom dialog box entitled "Dual voice" in the NVDA menu to select the secondary voice and setting of the Dual voice.
P.S. Could someone send this news to the NVDA addon user group, because my registration to that emailing list is not confirmed yet.

**Comment by josephsl** [comment]
Hi, I’ll take care of it and will close this issue. Thanks.
